samples/webassembly/Triangle/Triangle.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#define GLFW_INCLUDE_NONE
+#include <GLFW/glfw3.h>
+
+#ifndef __EMSCRIPTEN__
+#error Unsupported OS
+#endif
+#include <emscripten.h>
+
+#include <cassert>
+#include <regex>
+#include <stdio.h>
+
+#include <igl/IGL.h>
+#include <igl/opengl/webgl/Context.h>
+#include <igl/opengl/webgl/Device.h>
+
+#if defined(__cpp_lib_format)
+#include <format>
+#define IGL_FORMAT std::format
+#else
+#include <fmt/core.h>
+#define IGL_FORMAT fmt::format
+#endif // __cpp_lib_format
+
+std::string codeVS = R"(#version 300 es
```

Comment:
Can we use a constexpr string literal for this instead?
Say `constexpr const char*`.

We are using C++17, not 20, so this results in a global constructor.